SpamPkg/Core/Runtime/SpamResponderReport.c

```diff
@@ -0,0 +1,890 @@
+    Status = EFI_INVALID_PARAMETER;
+    goto Exit;
+  }
+
+  if (SpamResponderData->Signature != SPAM_RESPONDER_STRUCT_SIGNATURE) {
+    Status = EFI_UNSUPPORTED;
+    goto Exit;
+  }
+
+  if (SpamResponderData->VersionMajor > SPAM_REPSONDER_STRUCT_MAJOR_VER) {
+    Status = EFI_UNSUPPORTED;
+    goto Exit;
+  } else if ((SpamResponderData->VersionMajor == SPAM_REPSONDER_STRUCT_MAJOR_VER) &&
+            (SpamResponderData->VersionMinor > SPAM_REPSONDER_STRUCT_MINOR_VER)) {
+    Status = EFI_UNSUPPORTED;
+    goto Exit;
+  }
+
+  // Step 2: Check MM Entry code base and size to be inside the MMRAM region
+
+  AsmCpuid (CPUID_EXTENDED_FUNCTION, &MaxExtendedFunction, NULL, NULL, NULL);
+
+  if (MaxExtendedFunction >= CPUID_VIR_PHY_ADDRESS_SIZE) {
+    AsmCpuid (CPUID_VIR_PHY_ADDRESS_SIZE, &VirPhyAddressSize.Uint32, NULL, NULL, NULL);
+  } else {
+    VirPhyAddressSize.Bits.PhysicalAddressBits = 36;
+  }
+
+  MtrrValidBitsMask    = LShiftU64 (1, VirPhyAddressSize.Bits.PhysicalAddressBits) - 1;
+  MtrrValidAddressMask = MtrrValidBitsMask & 0xfffffffffffff000ULL;
+
+  MmRamBase = AsmReadMsr64 (MSR_IA32_SMRR_PHYSBASE);
+  MmrrMask = AsmReadMsr64 (MSR_IA32_SMRR_PHYSMASK);
+  // Extend the mask to account for the reserved bits.
+  MmrrMask |= 0xffffffff00000000ULL;
+  Length = ((~(MmrrMask & MtrrValidAddressMask)) & MtrrValidBitsMask) + 1;
+  MmBase = AsmReadMsr64 (MSR_IA32_SMBASE);
+  if (MmBase == 0) {
+    Status = EFI_SECURITY_VIOLATION;
+    goto Exit;
+  }
+
+  Status = Range1InsideRange2 (MmBase + SMM_HANDLER_OFFSET, SpamResponderData->MmEntrySize, MmRamBase, Length, &IsInside);
+  if (EFI_ERROR (Status) || !IsInside) {
+    Status = EFI_SECURITY_VIOLATION;
+    goto Exit;
+  }
+
+  if ((IMAGE_VALIDATION_DATA_HEADER*)(VOID*)SpamResponderData->MmSupervisorAuxBase == 0) {
+    Status = EFI_SECURITY_VIOLATION;
+    goto Exit;
+  }
+
+  Status = Range1InsideRange2 (SpamResponderData->MmSupervisorAuxBase, SpamResponderData->MmSupervisorAuxSize, MmRamBase, Length, &IsInside);
+  if (EFI_ERROR (Status) || !IsInside) {
+    Status = EFI_SECURITY_VIOLATION;
+    goto Exit;
+  }
+
+  if (((IMAGE_VALIDATION_DATA_HEADER*)SpamResponderData->MmSupervisorAuxBase)->HeaderSignature != IMAGE_VALIDATION_DATA_SIGNATURE) {
+    Status = EFI_SECURITY_VIOLATION;
+    goto Exit;
+  }
+
+  ZeroMem (DigestList, sizeof (DigestList));
+  Status = HashAndExtend (
+          SPAM_PCR_INDEX,
+          (VOID *)(UINTN)(SpamResponderData->MmSupervisorAuxBase),
+          (UINTN)SpamResponderData->MmSupervisorAuxSize,
+          DigestList
+          );
+  if (EFI_ERROR (Status)) {
+    goto Exit;
+  } else {
+    Status = EFI_NOT_FOUND;
+    for (Index = 0; Index < HASH_COUNT; Index++) {
+      if (DigestList[Index].count == 0) {
+        continue;
+      }
+
+      if (DigestList[Index].digests[0].hashAlg == TPM_ALG_SHA256) {
+        if (CompareMem (DigestList[Index].digests[0].digest.sha256, (VOID*)PatchPcdGetPtr (PcdAuxBinHash), SHA256_DIGEST_SIZE) != 0) {

```

Comment:
@Javagedes this is what i meant.

Reply:
Awesome - I missed this it seems. Thanks for pointing it out :)